cmd/estuary-shuttle/main.go

```diff
@@ -196,49 +339,31 @@ func main() {
 		},
 	}
 
-	app.Action = func(cctx *cli.Context) error {
-		ddir := cctx.String("datadir")
-
-		bsdir := cctx.String("blockstore")
-		if bsdir == "" {
-			bsdir = filepath.Join(ddir, "blocks")
-		} else if bsdir[0] != '/' && bsdir[0] != ':' {
-			bsdir = filepath.Join(ddir, bsdir)
-		}
-
-		wlog := cctx.String("write-log")
-		if wlog != "" && wlog[0] != '/' {
-			wlog = filepath.Join(ddir, wlog)
-		}
+	app.Commands = []*cli.Command{
+		{
+			Name:  "configure",
+			Usage: "Saves a configuration file to the location specified by the config parameter",
+			Action: func(cctx *cli.Context) error {
+				cfg := newDefaultConfig(cctx)
+				configuration := cctx.String("config")
+				cfg.Load(configuration) // Assume error means no configuration file exists
+				overrideSetOptions(cctx, cfg)
```

Comment:
Again, everything except `overrideSetOptions` should be outside of the `Action` logic